src/components/ValuePicker/types.ts

```diff
@@ -0,0 +1,58 @@
+type ValuePickerItem = {
+    value?: string;
+    label?: string;
+    description?: string;
+    text: string;
+    keyForList: string;
```

Comment:
I've checked what is passed as items to ValuePicker and it looks like `text` and `keyForList` shouldn't be required, I'm not sure if they are needed at all in this type.
https://github.com/Expensify/App/blob/e92ffbf97cfcce96ab6b6802225b6ace63953528/src/pages/workspace/WorkspaceNewRoomPage.js#L209-L216

Reply:
They cannot be unedfined for SelectionList

Reply:
But otherwise there will be TS errors during `WorkspaceNewRoomPage` migration, since only `value` and `label` are passed

Reply:
Ah, right, is creating a separated type for mapped values a good idea? WDYT?

Reply:
Could you clarify what you mean?

Reply:
Sure, in `ValueSelectorModal` I am mapping items to prepare them for SelectionList and the array has to have type that match the type in SelectionList
https://github.com/Expensify/App/pull/34654/files#:~:text=const%20%5BsectionsData%2C,selectedItem%5D)%3B
and when I remove `text` and `keyForList` from `ValuePickerItem` type it won't match SelectionList's type.
So should I create separate type for this mapped list from `ValueSelectorModal`?

Reply:
If it's not possible to make `text` and `keyForList` optional, then yeah. You can type `sectionsData` as `ListItem[]`
` const [sectionsData, setSectionsData] = useState<ListItem[]>([]);`
and add default values during mapping as it initially was

Reply:
Found solution with @blazejkustra - please check it @VickyStash